rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/100_knn_nested_search.yml

```diff
@@ -561,3 +584,32 @@ setup:
   - match: { hits.hits.0.inner_hits.nested.hits.hits.0.fields.nested.0.language.0: "FR" }
   - close_to: { hits.hits.0._score: { value: 0.0043, error: 0.0001 } }
 
+---
+"Test filter on sibling nested fields doesn't work":
+  - requires:
+      capabilities:
+        - method: POST
+          path: /_search
+          capabilities: [ knn_filter_on_nested_fields ]
+      test_runner_features: ["capabilities", "close_to"]
+      reason: "Capability for filtering on nested fields required"
+
+  - do:
+      search:
+        index: test
+        body:
+          _source: false
+          knn:
+            field: nested.vector
+            query_vector: [ -0.5, 90.0, -10, 14.8, -156.0 ]
+            filter:
+              nested:
+                path: nested2
+                query:
+                  bool:
+                    filter:
+                      - match:
+                          nested2.key: "category"
+                      - match:
+                          nested2.value: "domestic"
+  - match: { hits.total.value: 2}
```

Comment:
Here with the current implementation we can query nested sibling fields.

Reply:
I was sort of hoping that we could do something similar in the nested query since we look to the parent level. But it gets too complicated given the nested context?